### PR TITLE
Revamped buttons to fit spec

### PR DESCRIPTION
**Pull request by IanWoodard**
Overhauled the existing button component to better fit the new spec from Figma. This was quite the undertaking as there are a lot of possibilities with a lot of customization. I did my best to fit it all in without making it too verbose. I will attach some images of the various buttons so you will get a general idea of what they look like. I did not keep the button page since it was for testing, but if you want me to make a new testing page just let me know.
buttons1
buttons2
buttons3

### CODE REVIEW

File: src/components/common/AppBody.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@ import tw from 'twin.macro';
 export default styled.div`
   ${tw`bg-grey-50 w-full h-full text-white flex flex-col min-h-screen py-16`}
   font-family: 'Inter';
+  background-color: rgba(7, 14, 18, 1);
```
Comment (adamegyed) [inline src/components/common/AppBody.ts]: Shouldn't be here, should be in `src/index.css` as an attribute of `html` to set color on overscroll as well. Or, if this is only here to test the appearance of buttons against the spec background, let's remove this before merging.

**Review by adamegyed — COMMENTED** [review]
Could you re-add the buttons example page? It doesn't need to be mapped in the router in the commit, but it would help previewing the changes for this PR and future PRs that adjust the button designs.

**Comment by IanWoodard** [comment]
@adamegyed  Added the button example page at `/buttons` (will be removed before merging).

**Comment by adamegyed** [comment]
LGTM

File: src/components/common/Buttons.tsx [hunk 2]
```diff
@@ -1,172 +1,803 @@
-import React from 'react';
+import React, { ReactElement } from 'react';
 import styled from 'styled-components';
 import tw from 'twin.macro';
 import { classNames } from '../../util/ClassNames';
 import MigrateIcon from '../../assets/svg/migrate.svg';
```
Comment (adamegyed) [inline src/components/common/Buttons.tsx]: Can drop import here, as all icon dependencies have been shifted out to props.

**Review by adamegyed — APPROVED** [review]
Looks good. Remove the router reference to `/buttons`, but let's keep the page itself. I'll see if we can add the page to appear for routing only in dev mode with a new PR.

**Pull request merged**